src/components/RJST/components/Table/TableActions.tsx

```diff
@@ -41,25 +55,51 @@ export const TableActions = <T extends object>({
 				return;
 			}
 			if (typeof column.label === 'string' && column.label.startsWith('Tag:')) {
-				onColumnPreferencesChange(columns.filter((c) => c.key !== column.key));
+				onColumnPreferencesChange(
+					columns.filter((c) => c.key !== column.key),
+					'display',
+				);
 				return;
 			}
 			const newColumns = columns.map((c) =>
 				c.key === column.key ? { ...c, selected: !c.selected } : c,
 			);
-			onColumnPreferencesChange(newColumns);
+			onColumnPreferencesChange(newColumns, 'display');
 		},
 		[onColumnPreferencesChange, columns],
 	);
+
+	const handleDragEnd = (result: DropResult) => {
+		if (!result.destination || !analyticsState.featureFlags?.columnOrdering) {
+			return;
+		}
+
+		const newColumns = Array.from(columns);
+		const [movedColumn] = newColumns.splice(result.source.index, 1);
+		newColumns.splice(result.destination.index, 0, movedColumn);
+		const updatedColumns = newColumns.map((column, newIndex) => ({
+			...column,
+			index: newIndex,
+		}));
+
+		onColumnPreferencesChange?.(updatedColumns, 'reorder');
+	};
+
 	return (
 		<>
-			<IconButton
+			<Button
```

Comment:
nit: `ButtonWithTracking`?

Reply:
for this sepcific case we already send custom analytics, I would avoid sending them twice

https://github.com/balena-io-modules/ui-shared-components/blob/64d84d2a0bb48354956ca692f2709099222afb87/src/components/RJST/Lenses/types/table.tsx#L263